src/web/utils/errorHandler.tsx

```diff
@@ -0,0 +1,129 @@
+  hasError?: boolean;
+  message?: string;
+  stack?: string;
+  componentStack?: string;
+}
+
+export interface RenderedErrorProps {
+  errorId?: string;
+  errorHash?: string;
+  errorCount: number;
+  message?: string;
+  stack?: string;
+  componentStack?: string;
+}
+
+export function analyticsIdentifier(errorId?: string, errorHash?: string) {
+  if (!errorHash && !errorId) return 'No error identifier';
+  return errorHash
+    ? `error hash: ${errorHash}` // From backend
+    : `error ID: ${errorId}`; // Generated by frontend to make it possible to triage user reported errors.
+}
+
+export function ErrorView({
+  message,
+  errorId,
+  errorHash,
+}: {
+  message: string;
+  errorId?: string;
+  errorHash?: string;
+}) {
+  return (
+    <div className='error-content'>
+      <img alt='Error icon' src='uid2-logo.png' />
+      <div>Error</div>
+      <div>{message}</div>
+      <div>({analyticsIdentifier(errorId, errorHash)})</div>
+    </div>
+  );
+}
+export interface ErrorBoundaryPropType {
+  children: ReactNode;
+  [key: string]: unknown;
+}
+
+/** Create an Error Boundary around a given component.
+ * @param ErrorComponent - Custom error component used to display the error.
+ */
+export function errorHandler<P extends ErrorBoundaryPropType>(
+  ErrorComponent: React.ComponentType<RenderedErrorProps>
+) {
+  class ErrorHandlerWrapperComponent extends React.Component<P, ErrorState> {
+    ErrorComponent: React.ComponentType<RenderedErrorProps>; // eslint-disable-line react/no-unused-class-component-methods
+    constructor(props: P) {
+      super(props);
+      this.state = {
+        errorId: uuidv4() as string, // eslint-disable-line @typescript-eslint/no-unsafe-call
+        errorCount: 0,
+        hasError: false,
+        errorHash: undefined,
+      };
+      this.ErrorComponent = ErrorComponent; // eslint-disable-line react/no-unused-class-component-methods
+    }
+
+    componentDidCatch(error: Error, info: ErrorInfo) {
+      const errorId = uuidv4() as string; // eslint-disable-line @typescript-eslint/no-unsafe-call
+      const { message, stack } = error;
+      const { componentStack } = info;
+      const errorHash = error instanceof ApiError ? error.errorHash : undefined;
+      const newState = {
+        hasError: true,
+        errorId,
+        errorHash,
+        message,
+        stack,
+        componentStack,
+      };
+      this.setState(newState);
+    }
+
+    render() {
+      const { hasError, errorId, errorHash, errorCount, message, stack, componentStack } =
+        this.state;
+      const { children } = this.props;
+      if (hasError) {
+        return (
+          <this.ErrorComponent
+            errorId={errorId}
+            errorHash={errorHash}
+            errorCount={errorCount}
+            message={message}
+            stack={stack}
+            componentStack={componentStack}
+          />
+        );
+      }
+      return <React.Fragment key={errorCount}>{children}</React.Fragment>;
```

Comment:
what is errorCount for? is it always be 0?

Reply:
Yeah, this is unused right now. It's purpose is to re-mount the children if the user wants to retry any intermittent loading errors (user can retry intermittent failure in creatives). Let me remove it for now, and we can always add it back if design wants it that way.

Reply:
Removed ✅